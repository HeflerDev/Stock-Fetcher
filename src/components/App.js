import React from 'react';
import StocksForm from '../containers/StocksForm';
import StockList from '../containers/StockList';
import StockStatistics from '../containers/StockStatistics';

const App = () => (
  <div className="board">
    <div className="col-12 col-l-3">
      <StockStatistics />
    </div>
    <div className="col-12 col-l-9">
      <StocksForm />
      <StockList />
    </div>
  </div>
);

export default App;
