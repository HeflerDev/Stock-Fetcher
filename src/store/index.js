import { combineReducers, createStore } from 'redux';
import stocksReducer from '../reducers/stocksReducer';
import filterReducer from '../reducers/filterReducer';

const rootReducer = combineReducers({
  stocksReducer,
  filterReducer,
});

const store = createStore(rootReducer);
store.subscribe(() => console.log(store.getState));

export default store;
