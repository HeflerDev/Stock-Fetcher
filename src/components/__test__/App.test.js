// import { configure, shallow } from 'enzyme';
// import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import testRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import App from '../App';
import store from '../../store/index';

// configure({ adapter: new Adapter() });

describe('when loading the page', () => {
  const mockComp = (
    <Provider store={store}>
      <App />
    </Provider>
  );

  it('renders accordingly', () => {
    const tree = testRenderer
      .create(mockComp)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
