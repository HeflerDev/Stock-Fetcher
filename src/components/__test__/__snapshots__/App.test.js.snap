// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when loading the page renders accordingly 1`] = `
<div
  className="board"
>
  <div
    className="col-12 col-l-3"
  >
    <div
      className="stack statistics-container"
    >
      <h3>
        Actives
      </h3>
      <div
        className="queue center"
      >
        <p>
          Loading Stocks...
        </p>
      </div>
      <h3>
        Top Gainers
      </h3>
      <div
        className="queue center"
      >
        <p>
          Loading Stocks...
        </p>
      </div>
      <h3>
        Top Losers
      </h3>
      <div
        className="queue center"
      >
        <p>
          Loading Stocks...
        </p>
      </div>
    </div>
  </div>
  <div
    className="col-12 col-l-9"
  >
    <div
      className="stack"
    >
      <form
        className="board"
      >
        <h2
          className="col-12"
        >
          Search for Symbol or Name
        </h2>
        <label
          className="col-12 col-m-9"
          htmlFor="acronym"
        >
          <input
            className="queue"
            id="acronym"
            onChange={[Function]}
            placeholder="AAPL"
            type="text"
          />
        </label>
        <div
          className="queue center col-12 col-m-3 disabled"
        >
          <img
            alt="Search"
            src="magnifying-glass.svg"
          />
        </div>
      </form>
      
    </div>
    <div
      className="stack"
    />
  </div>
</div>
`;
